src/main/scala/com.snowplowanalytics/forex/Forex.scala

```diff
@@ -166,28 +171,28 @@ case class ForexLookupWhen(conversionAmount: Double, fromCurr: String, toCurr: S
    * Performs live currency lookup/conversion, no caching available
    * @return Money representation in target currency or OerResponseError object if API request failed
    */
-  def now: Either[OerResponseError, Money] = {
-    val timeStamp = DateTime.now
-    val from      = fx.client.getLiveCurrencyValue(fromCurr)
-    val to        = fx.client.getLiveCurrencyValue(toCurr)
-    if (from.isRight && to.isRight) {
-      // API request succeeds
-      val baseOverFrom       = from.right.get
-      val baseOverTo         = to.right.get
-      val fromCurrIsBaseCurr = (fromCurr == fx.config.baseCurrency)
-      val rate               = Forex.getForexRate(fromCurrIsBaseCurr, baseOverFrom, baseOverTo)
-      // Note that if `fromCurr` is not the same as the base currency,
-      // then we need to add the <fromCurr, toCurr> pair into the cache in particular,
-      // because only <baseCurrency, toCurr> were added earlier
-      if (!fx.client.caches.nowish.isEmpty && fromCurr != fx.config.baseCurrency) {
-        val Some(cache) = fx.client.caches.nowish
-        cache.put((fromCurr, toCurr), (timeStamp, rate))
+  def now: F[Either[OerResponseError, Money]] = {
+    val timeStamp = ZonedDateTime.now
```

Comment:
`ZonedDateTime.now` is side-effecting. In `traverse` we would have exactly same bug as we had in Scala Tracker.